source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -459,14 +499,21 @@ connections managed by the pool is at least minPoolSize.
 
 Populating the pool MUST NOT block any application threads. For example, it
 could be performed on a background thread or via the use of non-blocking/async
-I/O.
+I/O. Populating the pool MUST NOT be performed unless the pool is "ready".
+
+If an error is encountered while populating a connection, the pool MUST be
+cleared. The pool SHOULD notify the corresponding monitor that such an error
+occurred, if possible according to the driver's implementation.
 
 .. code::
 
-   wait until pendingConnectionCount < maxConnecting
+   wait until pendingConnectionCount < maxConnecting and pool is "ready"
    create connection
-   establish connection
-   mark connection as available
+   try:
+     establish connection
+     mark connection as available
+   except error:
+     clear pool
```

Comment:
This case also needs to check generation number to avoid clearing the pool multiple times. Also, are these errors supposed to mark the server unknown too in addition to clearing the pool?

Reply:
>This case also needs to check generation number to avoid clearing the pool multiple times.

Good point, fixed.

>Also, are these errors supposed to mark the server unknown too in addition to clearing the pool?

I left this as a "SHOULD", since I'm unsure if every driver is capable of implementing this, and if a driver doesn't implement it, SDAM will pick up the error nonetheless through the next `checkOut` and update the topology accordingly. I added some more explicit prose to make this clear and also added a line to the psuedocode. 

Additionally, to prevent repeated PoolCleared events in the case that the background thread doesn't update the topology, I added some prose requiring that paused pools don't emit PoolCleared events but do increment their generation count. I added rationale for this to the rationale section.